test/built-ins/Date/prototype/toString/non-date-receiver.js

```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2017 V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: #sec-date.prototype.tostring
+description: Date.prototype.toString throws a TypeError on non-Date receivers
+---*/
+
+assert.throws(TypeError, () => Date.prototype.toString());
```

Comment:
good catch! I hope we never give Date.prototype a [[DateValue]] internal slot anymore, even worse with a NaN value.